google/cloud/storage/client.py

```diff
@@ -1056,3 +1059,14 @@ def _item_to_hmac_key_metadata(iterator, item):
     metadata = HMACKeyMetadata(iterator.client)
     metadata._properties = item
     return metadata
+
+
+def _override_private_url_variables(api_endpoint):
+    blob._DOWNLOAD_URL_TEMPLATE = api_endpoint + u"/download/storage/v1{path}?alt=media"
+    blob._BASE_UPLOAD_TEMPLATE = (
+        api_endpoint + u"/upload/storage/v1{bucket_path}/o?uploadType="
+    )
+    blob._MULTIPART_URL_TEMPLATE = blob._BASE_UPLOAD_TEMPLATE + u"multipart"
+    blob._RESUMABLE_URL_TEMPLATE = blob._BASE_UPLOAD_TEMPLATE + u"resumable"
+    blob._API_ACCESS_ENDPOINT = api_endpoint
+    bucket._API_ACCESS_ENDPOINT = api_endpoint
```

Comment:
These URLs are used for Signature based requests like Signed URLs. These can be left alone because the method has a way to update the hostname because they don't use the same API (Download and Upload use JSON while Signature requests use XML API).